pkg/scheduler/service/bookkeeper_test.go

```diff
@@ -85,3 +92,116 @@ func TestBookkeeper(t *testing.T) {
 		require.NoError(t, reconRepo.RemoveReconciliation(recon.SchedulingID))
 	}
 }
+
+func TestBookkeeperParallel(t *testing.T) {
+	tests := []struct {
+		name        string
+		markOpsDone bool
+		customFunc  string
+		errMessage  string
+		errCount    int
+	}{
+		{name: "Mark two operations as orphan in multiple parallel threads", markOpsDone: false, customFunc: "markOrphanOperations", errMessage: "Bookkeeper failed to update status of orphan operation", errCount: 98},
+		{name: "Finish two operations in multiple parallel threads", markOpsDone: true, customFunc: "finishReconciliation", errMessage: "Bookkeeper failed to update cluster", errCount: 49},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			//make sure persistent db is empty
+			err := os.RemoveAll("./test/")
+			require.NoError(t, err)
+			//initialize WaitGroup
+			var wg sync.WaitGroup
+
+			//create mock database connection
+			dbConn := db.NewTestConnection(t)
+			//prepare inventory
+			inventory, err := cluster.NewInventory(dbConn, true, cluster.MetricsCollectorMock{})
+			require.NoError(t, err)
+
+			//add cluster to inventory
+			clusterState, err := inventory.CreateOrUpdate(1, &keb.Cluster{
```

Comment:
try to create a function or factory class to initialize the test dummy data, so that we can remove this big chunk of code in test body.

Actually, in `pkg/cluster/inventory_test.go:302`, you can already find a function which can create dummy cluster, the question is how we reference it.

Reply:
Will refactor this